### DISCUSSION TITLE
Changing table columns in CP for non-admins

### DISCUSSION THREAD

**Discussion by thaoms** [craftcms/cms#8562] [Ideas]
Is it possible to let non-admins change their table columns for entry index views. 
(Or admins change non-admins layout)

I can find this in the source code.

`{% set customizableSources = (sources is not empty and context == 'index' and currentUser.admin) %}
`

Does this have a specific reason? Some of our clients would like some different layout.

**Comment by brandonkelly** [comment]
It’s not currently possible so I’ll consider this a feature request.

**Comment by billythekid** [comment]
Perhaps this could be added as a permission? "Can edit own table columns" or something? Is there a workaround for this at all, other than setting everyone as admin?

**Comment by brandonkelly** [comment]
Right now the selected table columns are global, not user-specific. So if we allow anyone besides an admin to edit them, that would need to change first.

**Comment by billythekid** [comment]
@brandonkelly is there any way the admin's chosen order can be shown to all users who can view those elements once set until the ability to set on a per-user basis makes it in?

**Comment by brandonkelly** [comment]
If you mean the chosen column order, that’s already the case.

**Comment by billythekid** [comment]
@brandonkelly yeah I thought that was already the case! A client of mine was saying they weren't getting the saved columns in commerce product index/list page just the "defaults". I'll double-check with them this week and see what the story is. This was the only reason I raised it. Sorry for wasting your time. Ugh!
